File: ExampleServer.java
// example of sending an image from file image.jpg to a server at serveraddress:5000

import java.net.Socket;
import java.io.DataOutputStream;
import java.io.FileInputStream;
public class ExampleClient {
  public static void main(String[] args) {
    Socket outputSocket = new Socket("serveraddress", 5000);
    DataOutputStream dos = new DataOutputStream(outputSocket.getOutputStream());
    FileInputStream fis = new FileInputStream("image.jpg");
    
    byte[] data = new byte[65 /*size*/];
    fis.read(data);
    dos.write(data);
    
    dos.flush();
    dos.close();
    fis.close();
    outputSocket.close();
  }
}

File: ExampleServer.py
import socket
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind(("10.0.0.12", 5000))
s.listen(1000)
while True:
  try:
    client, _ = s.accept()
    img = open('img.jpg', 'wb')
    try:
      while True:
          data = client.recv(1024)
          if not data:
              break
          img.write(data)
    finally:
      img.close()
    print 'got image' #image written to img.jpg
  finally:
    client.close()
